amical/tests/test_extraction.py

```diff
@@ -22,21 +45,76 @@ def test_add_infos_simulated():
     assert infos["date-obs"] == hdr["DATE-OBS"]
     assert infos["telescop"] == hdr["TELESCOP"]
     assert "observer" not in infos  # Keys that are not in hdr should not be in infos
+    assert "observer" not in infos.hdr  # Keys that are not in hdr should still not be
 
 
-def test_add_infos_header_commentary():
+@pytest.mark.filterwarnings("ignore: Commentary cards")
+def test_add_infos_header_commentary(commentary_infos):
     # Make sure that _add_infos_header handles _HeaderCommentaryCards from astropy
 
+    # Convert everything to munch object
+    munch.munchify(commentary_infos)
+
+
+@pytest.mark.skipif(
+    ASTROPY_VERSION < Version("5.0rc"),
+    reason="Munch cannot handle commentary cards for Astropy < 5.0",
+)
+def test_commentary_infos_keep(commentary_infos):
+    assert "HISTORY" in commentary_infos.hdr
+
+
+@pytest.mark.skipif(
+    ASTROPY_VERSION >= Version("5.0rc"),
+    reason="Munch can handle commentary cards for Astropy 5.0+",
+)
+@pytest.mark.xfail(
+    ASTROPY_VERSION < Version("5.0rc"),
+    reason="AMICAL removes commentary cards from header with Astropy < 5.0",
+)
+def test_commentary_infos_drops(commentary_infos):
+    assert "HISTORY" in commentary_infos.hdr
+
+
+@pytest.mark.skipif(
```

Comment:
Same reasoning as above :)
```suggestion
@pytest.mark.xfail(
```

Reply:
Solved by last commit using xfail only and removing redundant tests.

Reply:
Solved by last commit using xfail only and removing redundant tests.